R/guide-axis.R

```diff
@@ -100,9 +103,26 @@ GuideAxis <- ggproto(
     line  = "axis.line",
     text  = "axis.text",
     ticks = "axis.ticks",
-    ticks_length = "axis.ticks.length"
+    minor = "axis.minor.ticks",
+    major_length = "axis.ticks.length",
+    minor_length = "axis.minor.ticks.length"
   ),
 
+  extract_key = function(scale, aesthetic, minor.ticks, ...) {
+    major <- Guide$extract_key(scale, aesthetic, ...)
+    if (!minor.ticks) {
+      return(major)
+    }
+    if (!is.null(major)) {
+      major$.type <- "major"
+    }
+    minor <- setdiff(scale$get_breaks_minor(), major$.value)
+    new_scale <- ggproto(NULL, scale, breaks = minor, get_labels = .no_labels)
+    minor <- Guide$extract_key(new_scale, aesthetic, ...)
```

Comment:
Is building up a whole new guide object really the best way to go about this? (asking in earnest). Wouldn't the above suggest the need for an additional method in `Guide`, or a switch in `Guide$extract_key()`

Reply:
This doesn't build a new guide, it uses the parental `extract_key()` method.
It's equivalent to `ggproto_parent(Guide, self)$extract_key()`, except guides are stateless, so we don't need that mechanism.
The copy-and-modify of the scale is a bit redundant here, but we might need it later for #5410. Anyway I've worked around t hat for now.